packages/circle-demo-webapp/app/hooks/useGetTransaction.ts

```diff
@@ -0,0 +1,45 @@
+import { GetTransactionInput } from '@circle-fin/developer-controlled-wallets';
+import { useState } from 'react';
+
+import { TransactionWithToken } from '~/lib/types';
+import { callGetFetch } from '~/lib/utils';
+
+interface UseGetTransactionResult {
+  error: Error | undefined;
+  transaction: TransactionWithToken | undefined;
+  isLoading: boolean;
+  getTransaction: (args: GetTransactionInput) => Promise<boolean>;
+  setTransaction: (transaction: TransactionWithToken | undefined) => void;
+}
+
+export const useGetTransaction = (): UseGetTransactionResult => {
+  const [transaction, setTransaction] = useState<TransactionWithToken | undefined>();
+  const [error, setError] = useState<Error | undefined>(undefined);
+  const [isLoading, setIsLoading] = useState<boolean>(false);
+
+  const getTransaction = async (args: GetTransactionInput) => {
+    setIsLoading(true);
+    setError(undefined);
+    try {
+      const res = await callGetFetch<{ transaction: TransactionWithToken }>(
+        `/api/getTransaction`,
+        args as unknown as Record<string, string>,
+      );
+      setTransaction(res.transaction);
+      return true;
+    } catch (err) {
+      setError(err as Error);
+      return false;
+    } finally {
+      setIsLoading(false);
+    }
+  };
+
+  return {
+    error,
+    isLoading,
+    transaction,
+    getTransaction,
+    setTransaction,
```

Comment:
Exposing function to modify internal state of a component is a really bad practice. What are you trying to achieve here with this approach?

Reply:
fixed